tools/convertUsageStatsLogFile.php

```diff
@@ -0,0 +1,570 @@
+                    }
+                } else {
+                    // consider this issue: https://github.com/pkp/pkp-lib/issues/6573
+                    // apache log files contain URL download/submissionId/galleyId, i.e. without third argument
+                    $submissionFileId = $galley->getData('submissionFileId');
+                    if (isset($args[2]) && $args[2] != $submissionFileId) {
+                        echo "Submission file with the ID {$submissionFileId} does not belong to the galley with the ID {$representationId}.\n";
+                        break;
+                    }
+                }
+                $submissionFile = Repo::submissionFile()->get($submissionFileId, $submissionId);
+                if (!$submissionFile) {
+                    echo "Submission file with the ID {$submissionFileId} does not exist in the submission with the ID {$submissionId}.\n";
+                    break;
+                }
+                if ($galley->getData('submissionFileId') != $submissionFileId) {
+                    echo "Submission file with the ID {$submissionFileId} does not belong to the represantation (galley or publication format) with the ID {$representationId}.\n";
+                    break;
+                }
+
+                // is this a full text or supp file
+                $genreDao = DAORegistry::getDAO('GenreDAO');
+                $genre = $genreDao->getById($submissionFile->getData('genreId'));
+                if ($genre->getCategory() != Genre::GENRE_CATEGORY_DOCUMENT || $genre->getSupplementary() || $genre->getDependent()) {
+                    $newEntry['assocType'] = Application::ASSOC_TYPE_SUBMISSION_FILE_COUNTER_OTHER;
+                } else {
+                    $newEntry['assocType'] = $assocType;
+                }
+                $newEntry['submissionId'] = $submissionId;
+                $newEntry['representationId'] = $representationId;
+                $newEntry['submissionFileId'] = $submissionFileId;
+                $newEntry['fileType'] = StatisticsHelper::getDocumentType($submissionFile->getData('mimetype'));
+                break;
+            case Application::ASSOC_TYPE_SUBMISSION:
+                $publicationId = null;
+                if (!isset($args[0])) {
+                    echo "Missing URL parameter.\n";
+                    break;
+                }
+                // If the operation is 'view' and the arguments count > 1
+                // the arguments must be: $submissionId/version/$publicationId.
+                // Else, it is not considered hier, as submission abstract count.
+                if ($op == 'view' && count($args) > 1) {
+                    if ($args[1] !== 'version') {
+                        echo "Wrong URL parameter. Expected 'verison' found {$args[1]}.\n";
+                        break;
+                    } elseif (count($args) != 3) {
+                        echo "Missing URL parameter.\n.\n";
```

Comment:
This could also be missing the publication parameter, right?

Reply:
Yes...

Reply:
Could you add that to the message?